commons/src/test/java/io/aiven/kafka/tieredstorage/commons/io/SegmentCompressionCheckerTest.java

```diff
@@ -0,0 +1,55 @@
+package io.aiven.kafka.tieredstorage.commons.io;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.file.Path;
+import org.apache.kafka.common.record.CompressionType;
+import org.apache.kafka.common.record.FileRecords;
+import org.apache.kafka.common.record.MemoryRecords;
+import org.apache.kafka.common.record.MemoryRecordsBuilder;
+import org.apache.kafka.common.record.TimestampType;
+import org.assertj.core.util.Files;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class SegmentCompressionCheckerTest {
```

Comment:
It would be great to have a test for cases where the first and second batches have different compression.

Reply:
With current approach (take only first batch) this wouldn't exercise any additional paths, though (ie. checker will return false) Unless you want to change the approach to compare first and last batch?

Reply:
I'm more about making the contract more explicit: to show only the first batch matters.

Reply:
I see! Make sense, adding these.